backend/tasks/user.py

```diff
@@ -0,0 +1,87 @@
+
+        # Check params
+        if enable is None or user is None:
+            return make_response(json.dumps({'message': 'Parameter \'enable\' required'}), 400)
+        if enable != "false" and enable != "true":
+            return make_response(json.dumps({'message': 'Parameter \'enable\' must have value true or false'}), 400)
+        if enable == "false" and queue is None:
+            return make_response(json.dumps({'message': 'Parameter \'queue\' required if disabling user'}), 400)
+
+        try:
+            with request.db.cursor(cursor_factory=RealDictCursor) if os.environ.get('IS_LOCAL') else request.db.cursor(DictCursor) as cur:
+
+                # Make sure user exists and can be enabled/disabled
+                cur.execute("SELECT \"user\".id, \"user\".username, \"user\".enabled FROM \"user\" WHERE \"user\".username=%s", (user,))
+                result = cur.fetchone()
+                if cur.rowcount == 0:
+                    return make_response(json.dumps({'message': f"Could not find user {user}"}), 404)
+                if result["enabled"] and enable == "true":
+                    return make_response(json.dumps({'message': f"User {user} already enabled"}), 400)
+                if not result["enabled"] and enable == "false":
+                    return make_response(json.dumps({'message': f"User {user} already disabled"}), 400)
+
+                # Update User
+                cur.execute("UPDATE \"user\" SET enabled=%s WHERE \"user\".username=%s", (enable, user))
+
+                # If user is being disabled clean up the builds, build triggers, mirrors, and build queue
+                if enable == "false":
+                    user_id = result["id"]
+                    cur.execute("SELECT id FROM Repository WHERE Repository.namespace_user_id=\'%s\'", (user_id,))
+                    repositories = cur.fetchall()
+                    if cur.rowcount > 0:
+                        # Convert list of rows into list of repo id's as strings, then join
+                        repo_ids = ", ".join([str(repo["id"]) for repo in repositories])
+                        cur.execute(f"DELETE FROM RepositoryBuild WHERE RepositoryBuild.repository_id IN ({repo_ids})")
+                        cur.execute(f"DELETE FROM RepositoryBuildTrigger WHERE RepositoryBuildTrigger.repository_id IN ({repo_ids})")
+                        cur.execute(f"DELETE FROM RepoMirrorConfig WHERE RepoMirrorConfig.repository_id IN ({repo_ids})")
+                        queue_prefix = "%s/%s/%%" % (queue, user)
+                        cur.execute("DELETE FROM QueueItem WHERE QueueItem.queue_name LIKE %s", (queue_prefix,))
+
+                request.db.commit()
+                return make_response(json.dumps({'message':'User updated successfully','user':user,'enabled':enable}), 200)
+        except Exception as e:
+            logger.exception("Unable to update enable status: " + str(e))
+            request.db.rollback()
```

Comment:
Do you need to explicitly roll back if you're using the pymysql context manager?
I thought when `with request.db.cursor()...`, it would be handled automatically.

Reply:
[From the source code](https://github.com/PyMySQL/PyMySQL/blob/main/pymysql/cursors.py#L56-L61) it doesn't look like it

Reply:
I don't think it's required given that no updates are made after the commit statement, and that DB connections are not being reused between requests.
If the commit statement is not executed, no changes will be made. 
The return statement should probably be outside of the "transaction" (the cursor() contextmanager)